index.js

```diff
@@ -1,250 +1,4 @@
-      prompt.value = prompt.input = value;
-
-      // if "autofill=show" render the prompt, otherwise stay "silent"
-      if (opts.autofill === 'show') {
-        await prompt.submit();
-      }
-    } else {
-      value = prompt.value = await prompt.run();
-    }
-
-    return value;
-  }
-
-  /**
-   * Use an enquirer plugin.
-   *
-   * ```js
-   * const Enquirer = require('enquirer');
-   * const enquirer = new Enquirer();
-   * const plugin = enquirer => {
-   *   // do stuff to enquire instance
-   * };
-   * enquirer.use(plugin);
-   * ```
-   * @name use()
-   * @param {Function} `plugin` Plugin function that takes an instance of Enquirer.
-   * @return {Object} Returns the Enquirer instance.
-   * @api public
-   */
-
-  use(plugin) {
-    plugin.call(this, this);
-    return this;
-  }
-
-  set Prompt(value) {
-    this._Prompt = value;
-  }
-  get Prompt() {
-    return this._Prompt || this.constructor.Prompt;
-  }
-
-  get prompts() {
-    return this.constructor.prompts;
-  }
-
-  static set Prompt(value) {
-    this._Prompt = value;
-  }
-  static get Prompt() {
-    return this._Prompt || require('./lib/prompt');
-  }
-
-  static get prompts() {
-    return require('./lib/prompts');
-  }
-
-  static get types() {
-    return require('./lib/types');
-  }
-
-  /**
-   * Prompt function that takes a "question" object or array of question objects,
-   * and returns an object with responses from the user.
-   *
-   * ```js
-   * const { prompt } = require('enquirer');
-   * const response = await prompt({
-   *   type: 'input',
-   *   name: 'username',
-   *   message: 'What is your username?'
-   * });
-   * console.log(response);
-   * ```
-   * @name Enquirer#prompt
-   * @param {Array|Object} `questions` Options objects for one or more prompts to run.
-   * @return {Promise} Promise that returns an "answers" object with the user's responses.
-   * @api public
-   */
-
-  static get prompt() {
-    const fn = (questions, ...rest) => {
-      let enquirer = new this(...rest);
-      let emit = enquirer.emit.bind(enquirer);
-      enquirer.emit = (...args) => {
-        fn.emit(...args);
-        return emit(...args);
-      };
-      return enquirer.prompt(questions);
-    };
-    utils.mixinEmitter(fn, new Events());
-    return fn;
-  }
-}
-
-utils.mixinEmitter(Enquirer, new Events());
-const prompts = Enquirer.prompts;
-
-for (let name of Object.keys(prompts)) {
-  let key = name.toLowerCase();
-
-  let run = options => new prompts[name](options).run();
-  Enquirer.prompt[key] = run;
-  Enquirer[key] = run;
-
-  if (!Enquirer[name]) {
-    Reflect.defineProperty(Enquirer, name, { get: () => prompts[name] });
-  }
-}
-
-const exp = name => {
-  utils.defineExport(Enquirer, name, () => Enquirer.types[name]);
-};
-
-exp('ArrayPrompt');
-exp('AuthPrompt');
-exp('BooleanPrompt');
-exp('NumberPrompt');
-exp('StringPrompt');
-
-module.exports = Enquirer;
+export default createEnquirerClass(nodeShims);
```

Comment:
Main change, using the shims in this entry point (NodeJS one). 
The entry point for Deno will be implemented last when the rest works, I can't test it until the changes have all been done anyway.